src/error.rs

```diff
@@ -67,7 +69,8 @@ impl From<ec_gpu_gen::EcError> for Error {
     }
 }
 
-impl error::Error for Error {}
+#[cfg(feature = "std")]
+impl std::error::Error for Error {}
 
 impl fmt::Display for Error {
```

Comment:
The qualified syntax easier to read here than needing to move back to the top of the file
```suggestion
impl core::fmt::Display for Error {
```